sdk/appcenter-data/src/main/java/com/microsoft/appcenter/data/Data.java

```diff
@@ -1014,6 +1014,10 @@ private TokenResult getCachedToken(String partitionName) {
             completeFuture(getInvalidPartitionDataException(partition), result);
             return true;
         }
+        if (documentId.isEmpty()) {
```

Comment:
This will crash if passing null. We also need to check for null.

Reply:
Don't we also need to check for document ids that contain invalid characters like `#`? @YiMysty ?